packages/lodestar/src/db/api/beacon/repositories/depositDataRoot.ts

```diff
@@ -29,28 +29,22 @@ export class DepositDataRootRepository extends Repository<number, Root> {
   }
 
   public async put(id: number, value: Root): Promise<void> {
-    if (!this.depositRootTree) {
-      await this.initTree();
-    }
+    const depositRootTree = await this.getDepositRootTree();
     await super.put(id, value);
-    this.depositRootTree[id] = value as TreeBacked<Root>;
+    depositRootTree[id] = value as TreeBacked<Root>;
   }
 
   public async batchPut(items: IKeyValue<number, Root>[]): Promise<void> {
-    if (!this.depositRootTree) {
-      await this.initTree();
-    }
+    const depositRootTree = await this.getDepositRootTree();
     await super.batchPut(items);
     for (const {key, value} of items) {
-      this.depositRootTree[key] = value as TreeBacked<Root>;
+      depositRootTree[key] = value as TreeBacked<Root>;
     }
   }
 
   public async getTreeBacked(depositIndex: number): Promise<TreeBacked<List<Root>>> {
-    if (!this.depositRootTree) {
-      await this.initTree();
-    }
-    const tree = this.depositRootTree.clone();
+    const depositRootTree = await this.getDepositRootTree();
```

Comment:
Same logic but by calling this function tsc can guarantee `depositRootTree` is defined